installers/msi-language/StateDeploy/CustomAction.cs

```diff
@@ -80,108 +80,58 @@ public static ActionResult InstallStateTool(Session session, out string stateToo
+            {
+                session.Log(string.Format("Attempting to login as user: {0}", username));
+                authCmd = stateToolPath + " auth" + " --username " + username + " --password " + password;
+            }
 
-                while (!proc.HasExited)
-                {
-                    try
-                    {
-                        // This is just hear to throw an InstallCanceled Exception if necessary
-                        Status.ProgressBar.Increment(session, 0);
-                        Thread.Sleep(200);
-                    }
-                    catch (InstallCanceledException)
-                    {
-                        session.Log("Caught install cancelled exception");
-                        ActiveState.Process.KillProcessAndChildren(proc.Id);
-                        output = "process got interrupted.";
-                        return ActionResult.UserExit;
-                    }
-                }
-                proc.WaitForExit();
+            string output;
+            Status.ProgressBar.StatusMessage(session, "Authenticating...");
+            ActionResult runResult = ActiveState.Command.Run(session, authCmd, ActiveState.Shell.Cmd, out output);
+            if (runResult.Equals(ActionResult.UserExit))
+            {
+                // Catch cancel and return
+                return runResult;
+            }
+            else if (runResult == ActionResult.Failure)
+            {
+                Record record = new Record();
+                session.Log(string.Format("Output: {0}", output));
+                var errorOutput = FormatErrorOutput(output);
+                record.FormatString = string.Format("Failed with error:\n{0}", errorOutput);
 
-                var exitCode = proc.ExitCode;
-                session.Log(String.Format("process returned with exit code: {0}", exitCode));
-                proc.Close();
-                if (exitCode != 0)
-                {
-                    outputBuilder.Append('\x00');
-                    outputBuilder.AppendFormat("Process returned with exit code: {0}", exitCode);
-                    output = outputBuilder.ToString();
-                    session.Log("returning due to return code - error");
-                    ActiveState.RollbarHelper.Report(
-                        string.Format("failed due to return code: {0} - error", exitCode), 
-                        new Dictionary<string, object> { { "output", output }, { "cmd", cmd } }
-                    );
-                    return ActionResult.Failure;
-                }
+                session.Message(InstallMessage.Error | (InstallMessage)MessageBoxButtons.OK, record);
+                return runResult;
             }
-            catch (Exception objException)
+            // The auth command did not fail but the username we expected is not present in the output meaning
+            // another user is logged into the State Tool 
+            else if (!output.Contains(username))
```

Comment:
This is strange behavior by the state tool. If we cannot login with a requested user, the `state auth` command should fail IMO...

I created a story for this: https://www.pivotaltracker.com/story/show/173984323 and added an AC to ensure that the change would still work with the language MSI...